gradle-plugin/test/proguard/gradle/ProguardCacheRelocateabilityIntegrationTest.groovy

```diff
@@ -0,0 +1,164 @@
+package proguard.gradle
+
+import org.apache.commons.io.FileUtils
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class ProguardCacheRelocateabilityIntegrationTest extends Specification {
+    @Rule TemporaryFolder temporaryFolder
+    @Rule TemporaryFolder cacheFolder
+
+    @Unroll
+    def "proguard task can be relocated"() {
+        def cacheDir = cacheFolder.newFolder()
+        def originalDir = temporaryFolder.newFolder()
+        def fixture = new File(getClass().classLoader.getResource('spring-boot').path)
+        FileUtils.copyDirectory(fixture, originalDir)
+        writeBuildGradle(originalDir)
+        writeSettingsGradle(originalDir, cacheDir)
+
+        def relocatedDir = temporaryFolder.newFolder()
+        FileUtils.copyDirectory(fixture, relocatedDir)
+        writeBuildGradle(relocatedDir)
+        writeSettingsGradle(relocatedDir, cacheDir)
+
+        when:
+        def result = GradleRunner.create()
+            .forwardOutput()
+            .withArguments('proguard', '--build-cache')
+            .withPluginClasspath()
+            .withProjectDir(originalDir)
+            .build()
+
+        def result2 = GradleRunner.create()
+            .forwardOutput()
+            .withArguments('proguard', '--build-cache')
+            .withPluginClasspath()
+            .withProjectDir(relocatedDir)
+            .build()
+        then:
+        result.output =~ "SUCCESSFUL"
+        result2.output =~ "SUCCESSFUL"
+        result2.output =~ "Task :proguard FROM-CACHE"
```

Comment:
I would add a check that the outputs are present in the relocated dir.  What we want to avoid is someone making changes to the outputs such that they are not included in the cache entry, which could result in correct outputs when the task is executed but only partial outputs when they come from cache.

I would also add a check such that when an input is changed, the result is _not_ pulled from cache.  We don't need to be exhaustive, but we want to at least prove that it gets a cache hit when it should and that it gets a cache miss when it should.

There should be no need to check for `SUCCESSFUL` in the output.  `build()` always checks for success (as opposed to `buildAndFail()` which always checks for failure).

Lastly, I would use `result.task(':proguard') == TaskOutcome.FROM_CACHE` here to be resilient against possible future changes in task output formatting.